JSIL.Libraries/Sources/JSIL.Core.js

```diff
@@ -10732,4 +10739,207 @@ JSIL.Array.IndexOf = function (array, startIndex, count, value) {
   }
 
   return -1;
-};
\ No newline at end of file
+};
+
+JSIL.MethodPointerInfo = function(typeObject, name, signature, isStatic, isVirtual, methodGenericParameters) {
+  this.TypeObject = typeObject;
+  this.Name = name;
+  this.Signature = signature;
+  this.IsStatic = isStatic;
+  this.MethodGenericParameters = methodGenericParameters || null;
+  this.IsVirtual = isVirtual;
+  this.MethodInfoResolved = false;
+  this.MethodInfo = null;
+
+  var useGenericSuffix = this.MethodGenericParameters !== null && this.MethodGenericParameters.length > 0;
+
+  this.NameWithGenericSuffix = useGenericSuffix ? (name + "$b" + this.MethodGenericParameters.length) : name;
+}
+
+JSIL.MethodInfoMethodPointerInfo = function(methodInfo) {
```

Comment:
Since this type has no methods of its own, please change this to a static method of JSIL.MethodPointerInfo, like ```JSIL.MethodPointerInfo.FromMethodInfo(mi)```.